### PR TITLE
Add formatting as workflow (as requested)

### PR DESCRIPTION
**Pull request by tobezdev**
**This PR is not a domain creation request.**
Reference pull #13966 

Version 2 of adding [Prettier](https://prettier.io/) formatting to is-a-dev/register!
New in this version:
* moved manual workflow to automatic:
  - [+] `.github/workflows/formatting.yml` - create the workflow
  - {~] `package.json` - added configuration for the formatter
  - [~] `package-lock.json` - added script to run formatting on-change
  - [+] `.prettierrc` - add configuration for formatting to run with
  - [+] `.prettierignore` - add a list of files for formatting to ignore

**Please request changes on `.prettierignore` before merging this to ensure the formatter only edits the correct files. You may also want to check/test the workflow to ensure it runs as intended.**

### CODE REVIEW

**Review by MaskDuck — COMMENTED** [review]
thank you for your second attempt at this, I highly value that!

File: .github/workflows/formatting.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: 'File Formatting with Prettier'
+
+on:
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: '00 8 * * *'
```
Comment (wdhdev) [inline .github/workflows/formatting.yml:8]: Please add a comment for how long this is

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+# Ignore all markdown files as these are broken by formatting (working on a fix for this)
+*.md
+
+# I have included everything in this document, just comment files & directories
+# you wish to have formatted - remove the hash from the line you wish to un-ingore
+
+#.github/
+#domains/
+#mesia/
```
Comment (wdhdev) [inline .prettierignore:11]:
```suggestion
#media/
```

File: .prettierignore [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+#node_modules/
```
Comment (wdhdev) [inline .prettierignore:12]:
```suggestion
```

File: package-lock.json
Comment (wdhdev) [inline package-lock.json:1]: I don’t believe we need a package-lock.json

File: .github/workflows/formatting.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  format:
+    if: github.repository == 'is-a-dev/register'
+    runs-on: ubuntu-latest
+    permissions: [read-all, write-all]
+    steps:
+      run: npx prettier domains/* --check
+      if github.working-directory == 'domains/':
+        run: npx prettier domains/* --write
+
```
Comment (wdhdev) [inline .github/workflows/formatting.yml:19]:
```suggestion
```
Empty line

**Review by wdhdev — CHANGES_REQUESTED** [review]

File: .github/workflows/formatting.yml (hunk 1 above)
Comment (TobezDev) [reply .github/workflows/formatting.yml:8]: It's 08:00am every morning, indefinitely.
Did you want it in the file?

File: package-lock.json
Comment (TobezDev) [reply package-lock.json:1]: You already had a `package-lock.json` file, I just updated it to include the `prettier` module.

File: .github/workflows/formatting.yml (hunk 1 above)
Comment (wdhdev) [reply .github/workflows/formatting.yml:8]:
```suggestion
    - cron: '0 0 * * *'
```

**Review by wdhdev — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (TobezDev) [reply package-lock.json:1]: I see one directly on https://github.com/is-a-dev/register.
I also forked the repo directly from Github, without adding the rest of the package-lock.json.
Not sure what's going on here, I'll take a proper look later when I'm at my computer.

Comment (wdhdev) [reply package-lock.json:1]: https://github.com/is-a-dev/register/blob/main/package-lock.json

**Review by wdhdev — COMMENTED** [review]
Can you ensure this works by running it in a test repo?

**Comment by tobezdev** [comment]
Sure @wdhdev, I'll do it later on today :)

**Review by MaskDuck — CHANGES_REQUESTED** [review]

**Comment by wdhdev** [comment]
> Sure @wdhdev, I'll do it later on today :)

Have you done this yet? If so, please send me the action run.

**Comment by omsenjalia** [comment]
> Sure @wdhdev, I'll do it later on today :)

Any updates?

**Comment by CuteDog5695** [comment]
@tobezdev Any updates?

**Comment by tobezdev** [comment]
Yes. I have managed to recreate the workflow on a private test repo. The formatting seemed to work fine, but I'm trying to troubleshoot a check failure (though it seems to work fine) 
I'll add another comment here when I find a fix to this.

**Comment by CuteDog5695** [comment]
Any updates?

**Comment by tobezdev** [comment]
> Any updates?

Going to be completely honest with you, I completely forgot about the existence of this PR. I've been very busy with personal stuff, so apologies for that.
Could someone remind me where we were with this?
Thanks.

**Comment by CuteDog5695** [comment]
> > Any updates?
> 
> Going to be completely honest with you, I completely forgot about the existence of this PR. I've been very busy with personal stuff, so apologies for that. Could someone remind me where we were with this? Thanks.

So from what I see, you were trying to troubleshoot a check failure, then MaskDuck tried out your workflow and pointed out that any nonzero exit code will result in the failure of the workflow as a flow, therefore stopping checks. That's the last update I've seen on this PR.
If you have no idea on what I'm saying, you could read the 2 messages that I've summarized that are above my follow up or you can even check your repo to see what you did there the last time.

**Comment by wdhdev** [comment]
I'm going to go ahead and close this. I appreciate the efforts however it is too complicated and would require thorough review from our team which we do not have the resources for.

**Pull request closed without merging**